module_checksum_upgrade/models/module.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# Copyright 2018 ACSONE SA/NV.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import json
+import logging
+import os
+
+from openerp import api, models
+from openerp.modules.module import get_module_path
+
+_logger = logging.getLogger(__name__)
+try:
+    from checksumdir import dirhash
+except ImportError:
+    _logger.debug('Cannot `import checksumdir`.')
+
+PARAM_INSTALLED_CHECKSUMS = \
+    'module_checksum_upgrade.installed_checksums'
+PARAM_EXCLUDED_EXTENSIONS = \
+    'module_checksum_upgrade.checksum_excluded_extensions'
+
+
+class Module(models.Model):
+    _inherit = 'ir.module.module'
+
+    @api.multi
+    def _get_checksum_dir(self):
+        self.ensure_one()
+
+        excluded_extensions = self.env["ir.config_parameter"].get_param(
+            PARAM_EXCLUDED_EXTENSIONS,
+            'pyc,pyo,pot',
```

Comment:
From https://github.com/OCA/server-tools/pull/1176#issuecomment-371625731 I got the idea that we should add here some more: `js,css,less,sass,scss`. Upgrades in those files land with a simple F5, and renames on them would land in an XML after all, so we can skip them safely too.

Reply:
These should be restricted to the static directory. I update the roadmap.

Reply:
Can they actually be found elsewhere?

Reply:
you never know what developers can invent, safety first

Reply:
Come on... This is just a sane default, if you install an addon that implements Odoo in JS, you can add this system parameter for sure :laughing: